### PR TITLE
Fix problems with OutOfBound state creation

### PR DESCRIPTION
**Pull request by vladyslav-dubina**
Example test: all of parserUnroll tests or issue692-bmv2.p4.
Solves problems related to incorrect generation of verify functions and creation of OutOfBound state

### CODE REVIEW

**Review by fruffy — APPROVED** [review]

File: midend/parserUnroll.h [hunk 1]
```diff
@@ -197,15 +197,22 @@ class RewriteAllParsers : public Transform {
...
                     IR::ID("StackOutOfBounds"))));
             IR::IndexedVector<IR::StatOrDecl> components;
-            components.push_back(new IR::MethodCallStatement(
-                new IR::MethodCallExpression(new IR::PathExpression(IR::ID("verify")), arguments)));
-            auto* outOfBoundsState = new IR::ParserState(IR::ID(outOfBoundsStateName), components,
-                nullptr);
+            components.push_back(new IR::MethodCallStatement(new IR::MethodCallExpression(
+                IR::Type::Void::get(),
+                new IR::PathExpression(
+                    new IR::Type_Method(IR::Type::Void::get(), new IR::ParameterList(), "*method"),
```
Comment (fruffy) [inline midend/parserUnroll.h:209]: The type of the method does not quite match. You add an empty parameter list here, but verify has the type parameters bool and error.

**Review by fruffy — CHANGES_REQUESTED** [review]

**Review by fruffy — APPROVED** [review]

**Comment by mbudiu-vmw** [comment]
I didn't have time to review this

**Comment by fruffy** [comment]
Ah I apologize. This one was fairly small and benign, so I just went ahead and merged it.

File: testdata/p4_16_samples_outputs/parser-unroll/parser-unroll-test1-midend.p4 [hunk 2]
```diff
@@ -43,6 +43,7 @@ parser MyParser(packet_in packet, out headers hdr, inout metadata meta, inout st
     @name("MyParser.index") int<32> index_0;
     state stateOutOfBound {
         verify(false, error.StackOutOfBounds);
+        transition reject;
```
Comment (mbudiu-vmw) [inline testdata/p4_16_samples_outputs/parser-unroll/parser-unroll-test1-midend.p4:46]: this is unreachable, verify(false) will always transition to reject.
https://p4.org/p4-spec/docs/P4-16-v1.2.2.html#sec-verify
So this change is really unnecessary.

**Pull request merged**